pkg/controller/jaeger/role.go

```diff
@@ -0,0 +1,52 @@
+package jaeger
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/inventory"
+)
+
+func (r *ReconcileJaeger) applyRoles(jaeger v1alpha1.Jaeger, desired []rbacv1.Role) error {
+	opts := client.MatchingLabels(map[string]string{
+		"app.kubernetes.io/instance":   jaeger.Name,
+		"app.kubernetes.io/managed-by": "jaeger-operator",
+	})
+	list := &rbacv1.RoleList{}
+	if err := r.client.List(context.Background(), opts, list); err != nil {
+		return err
+	}
+
+	logFields := log.WithFields(log.Fields{
+		"namespace": jaeger.Namespace,
+		"instance":  jaeger.Name,
+	})
+
+	inv := inventory.ForRoles(list.Items, desired)
+	for _, d := range inv.Create {
+		logFields.WithField("account", d.Name).Debug("creating role")
```

Comment:
The same comment as before:

* remane account to role and move it do fields. 
* remove role from debug.

Reply:
Good catch on the naming problem, I just fixed that.

The role is part of the loop, it can't be moved to fields.